ripe/atlas/tools/commands/report.py

```diff
@@ -67,18 +73,37 @@ def add_arguments(self):
                  "selected option.  Note that if you opt for aggregation, no "
                  "output will be generated until all results are received."
         )
+        self.parser.add_argument(
+            "--start-time",
+            type=ArgumentType.datetime,
+            help="The start time of the report."
+        )
+        self.parser.add_argument(
+            "--stop-time",
+            type=ArgumentType.datetime,
+            help="The stop time of the report."
+        )
 
-    def get_probes(self):
+    def _get_request(self):
+
+        kwargs = {"msm_id": self.arguments.measurement_id}
         if self.arguments.probes:
-            return [int(i) for i in self.arguments.probes.split(",")]
-        return []
+            kwargs["probe_ids"] = ",".join([str(_) for _ in self.arguments.probes])
+        if self.arguments.start_time:
+            kwargs["start"] = self.arguments.start_time
+        if self.arguments.stop_time:
+            kwargs["stop"] = self.arguments.stop_time
+
+        if "start" in kwargs or "stop" in kwargs:
+            return AtlasResultsRequest(**kwargs)
+        return AtlasLatestRequest(**kwargs)
 
     def run(self):
 
         self.payload = ""
         pk = self.arguments.measurement_id
         measurement_exists, detail = AtlasRequest(
-            url_path=self.URLS["detail"].format(pk)).get()
+            url_path=self.DETAIL_URL.format(pk)).get()
 
```

Comment:
Oh that's so much better.  Ok I'll modify my stuff to use this.

Just one question though: it's `meta_data` and not `meata_data`, right?